### PR TITLE
Check for Java 17 toolchain and fail if not found

### PR DESCRIPTION
**Pull request by bogdandrutu**
You cannot build unless java17 is installed, see https://github.com/open-telemetry/opentelemetry-java/blob/main/buildSrc/build.gradle.kts#L69

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -5,9 +5,9 @@ plugins {
   id("com.diffplug.spotless") version "6.25.0"
 }
 
-if (!JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17)) {
+if (JavaVersion.current() != JavaVersion.VERSION_17) {
```
Comment (jack-berg) [inline buildSrc/build.gradle.kts:8]: Requiring that the current version is java 17 is a bit stricter than we need. We really need to make sure that gradle can [autodetect](https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection) java version 17 on the system.

What about the following instead:
```
if (!hasLauncherForJavaVersion(17)) {
  throw GradleException(
    "JDK 17 is required to build and gradle was unable to detect it on the system.  " +
      "Please install it and see https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection " +
      "for details on how gradle detects java toolchains."
  )
}

fun hasLauncherForJavaVersion(version: Int): Boolean {
  return try {
    javaToolchains.launcherFor { languageVersion = JavaLanguageVersion.of(version) }.get()
    true
  } catch (e: Exception) {
    false
  }
}
```
I tested this locally and `hasLauncherForJavaVersion` successfully returned true for versions I had installed, and false for versions I did not.

**Comment by breedx-splk** [comment]
Yeah, we don't want to break users who are already on 21.

**Review by jack-berg — APPROVED** [review]
I've pushed a commit which reflects this recommendation: https://github.com/open-telemetry/opentelemetry-java/pull/6303#discussion_r1530671032

Updated the PR title

**Comment by jord1e** [comment]
Hi, here from the release notes.

You can actually configure Gradle to install the JDK when it is not found on the local system, this is a better experience than erroring and was the default up to Gradle 7.x or so.
You can keep the check in this PR for sanity.

Just edit `settings.gradle(.kts)` [NOT` build.gradle(.kts)`] and add this one line plugin. Here is a snippet:

```
plugins {
    id 'org.gradle.toolchains.foojay-resolver-convention' version '0.8.0'
}

rootProject.name = 'xxx'
```

See the official docs: https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning

**Pull request merged**